spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowTaskConfiguration.java

```diff
@@ -156,4 +158,10 @@ public TaskDeploymentReader taskDeploymentReader(TaskDeploymentRepository reposi
 	public JdbcTaskBatchDao taskBatchDao(DataSource dataSource) {
 		return new JdbcTaskBatchDao(dataSource);
 	}
+
+	@Bean
+	@ConditionalOnMissingBean
```

Comment:
The solution in this PR requires that we create a `DataflowTaskExecutionQueryDao` when Tasks are disabled, which is not the desired goal. 

Please look at PR https://github.com/spring-cloud/spring-cloud-dataflow/pull/5971 .  This PR resolves the problem without creating a `DataflowTaskExecutionQueryDao` when tasks are disabled.

Reply:
5971 fixes this issue in a more elegant way.